File: January 2026 GFG SOLUTION/January-16.java
class Solution {
    public int minMen(int arr[]) {
        int n = arr.length;
        List<int[]> v = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (arr[i] != -1) {
                v.add(new int[]{Math.max(0, i - arr[i]), Math.min(n - 1, i + arr[i])});
            }
        }
        if (v.isEmpty()) return -1;
        v.sort((a, b) -> a[0] - b[0]);
        int cnt = 0, pos = -1, i = 0;
        while (pos < n - 1) {
            if (i >= v.size() || v.get(i)[0] > pos + 1) return -1;
            int reach = pos;
            while (i < v.size() && v.get(i)[0] <= pos + 1) reach = Math.max(reach, v.get(i++)[1]);
            cnt++;
            pos = reach;
        }
        return cnt;
    }
}

File: January 2026 GFG SOLUTION/January-16.py
class Solution:
    def minMen(self, arr):
        n = len(arr)
        v = []
        for i in range(n):
            if arr[i] != -1:
                v.append((max(0, i - arr[i]), min(n - 1, i + arr[i])))
        if not v: return -1
        v.sort()
        cnt, pos, i = 0, -1, 0
        while pos < n - 1:
            if i >= len(v) or v[i][0] > pos + 1: return -1
            reach = pos
            while i < len(v) and v[i][0] <= pos + 1:
                reach = max(reach, v[i][1])
                i += 1
            cnt += 1
            pos = reach
        return cnt
